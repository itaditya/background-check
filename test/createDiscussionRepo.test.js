const createDiscussionRepo = require('../lib/github-api/createDiscussionRepo')

test('createDiscussionRepo is working', async () => {
  const github = {
    repos: { createForOrg: jest.fn() }
  }

  // Call your function, which internally calls your above mock function
  await createDiscussionRepo(github, { appInstallerName: 'test-org' })

  // Test that your mock function has been called
  expect(github.repos.createForOrg).toHaveBeenCalledWith({
    org: 'probot-background-check',
    name: 'test-org-discussions',
    description: 'Repo to have discussions about toxic users',
    // private: true,
    auto_init: true
  })

  // Test that your mock function has been called with the right arguments
  expect(github.repos.createForOrg.mock.calls).toMatchSnapshot()
})
