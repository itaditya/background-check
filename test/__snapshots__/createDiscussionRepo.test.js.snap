// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createDiscussionRepo is working 1`] = `
Array [
  Array [
    Object {
      "auto_init": true,
      "description": "Repo to have discussions about toxic users",
      "name": "test-org-discussions",
      "org": "probot-background-check",
    },
  ],
]
`;
